### PR TITLE
Niko/mvc

### PR DESCRIPTION
**Pull request by niko79542**
Finished refactoring for more of an mvc style, maintainable app.  Let me know what you think.

### CODE REVIEW

File: src/db/resolvers/comment_resolver.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    updateComment(_, args) {
+      return "success";
+    },
+    deleteComment(_, args) {
+      return "success";
+    },
+    voteComment(_, args) {
+      return "success"
+    }
+  }
```
Comment (jay-ithiel) [inline src/db/resolvers/comment_resolver.js]: Is this a skeleton?

File: src/db/resolvers/post_resolver.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+`;
+
+const postResolver = {
+  Query: {
+    getAllPosts(_, args) {
+      return Post.findAll({order: [['hot', 'ASC']]});
+    },
+    getPostReplies(_, args) {
+      return client.sendAsync(args.message, [args.params], (result) => {return result}); // e.g. message: "get_content_replies", params: ["steemit", "firstpost"]
+    },
```
Comment (jay-ithiel) [inline src/db/resolvers/post_resolver.js]: What if `getAllPosts` or `getPostReplies` fails?

File: src/db/resolvers/post_resolver.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    updatePost(_, args) {
+      return "success";
+    },
+    deletePost(_, args) {
+      return "success";
+    },
+    votePost(_, args) {
+      return "success"
+    }
```
Comment (jay-ithiel) [inline src/db/resolvers/post_resolver.js]: could we console.log the args before returning so if there is an error we can see what args we are getting/not getting

Comment (jay-ithiel) [reply src/db/resolvers/post_resolver.js]: or maybe even do a sanity check on args here and return success or error based on sanity check result

File: src/db/schema.js [hunk 4]
```diff
@@ -1,66 +1,29 @@
...
+  createNewPost: String
+  updatePost: String
+  deletePost: String
+  votePost: String
+  createNewComment: String
+  updateComment: String
+  deleteComment: String
+  voteComment: String
+  userLogin: String
 }
```
Comment (jay-ithiel) [inline src/db/schema.js]: the names sound like they're functions but they're set to a String. 
Are they going to be gql queries in a string json format?

File: src/db/schema.js [hunk 5]
```diff
@@ -17,7 +17,7 @@ type Query {
   updateComment: String
   deleteComment: String
   voteComment: String
-  userLogin: String
+  getLoginLink: String
 }
```
Comment (jay-ithiel) [inline src/db/schema.js]: getLoginLink sounds like a function as well, but it's type String so it's a bit confusing

File: src/db/schema.js [hunk 6]
```diff
@@ -21,9 +21,13 @@ type Query {
...
 `;
 
+// Hey.  the type defs (endpoint signatures) and resovlers need to be kept in the same file or graphql throws an error.
+// Think we should just merge these in the schema file for now...
+const resolvers = merge(postResolver, commentResovler, userResolver);
```
Comment (jay-ithiel) [inline src/db/schema.js:26]: sounds good

File: src/db/connectors.js [hunk 7]
```diff
@@ -1,7 +1,7 @@
 const casual = require('casual');
 const _ = require('lodash');
 
-const batchUpdate = require('../database_update/batch');
+const batchUpdate = require('../services/db_update');
```
Comment (jay-ithiel) [inline src/db/connectors.js:5]: could we give `batchUpdate` a more descriptive/specific name of what it's doing? It's very vague/general right now

**Review by jay-ithiel — CHANGES_REQUESTED** [review]
left some comments

File: src/db/schema.js [hunk 8]
```diff
@@ -9,7 +9,7 @@ const typeDefs = `
 type Query {
   getAllPosts: [Post]
   getPostReplies(author: String, permlink: String): String
-  createNewPost: String
+  createPost: String
   updatePost: String
```
Comment (jay-ithiel) [inline src/db/schema.js]: renamed to createPost since creating a post implies it's a new post

File: src/db/resolvers/post_resolver.js [hunk 9]
```diff
@@ -49,9 +53,21 @@ type Post {
...
+  }
+
+  type Mutation {
+    createPost: Post
+    updatePost(postId: String!): Post
+    deletePost(postId: String!): Post
+    votePost(postId: String!): Post
+  }
+
   type Post {
```
Comment (jay-ithiel) [inline src/db/resolvers/post_resolver.js]: We should separate queries from mutations

Comment (jay-ithiel) [reply src/db/resolvers/post_resolver.js]: when we votePost, are we updating the post? would it be possible to merge votePost into updatePost? or are they doing different things

File: src/db/resolvers/post_resolver.js [hunk 10]
```diff
@@ -73,30 +89,31 @@ const postTypeDefs = `
...
+      return 'success'
     }
   }
 };
 
-module.exports = {
-  postTypeDefs,
-  postResolver,
-}
+module.exports = makeExecutableSchema({ typeDefs, resolvers });
```
Comment (jay-ithiel) [inline src/db/resolvers/post_resolver.js]: I made an example of handling success/failures under `createPost`

also, it's bad practice to use double quotes in js. unless it's absolutely necessary, double quotes should be reserved for html

File: src/db/resolvers/post_resolver.js (hunk 9 above)
Comment (niko79542) [reply src/db/resolvers/post_resolver.js]: Hey - Good points.  updatePost will be edits to the body/title of the post.  VotePost will be upvote/downvote by other users

Comment (jay-ithiel) [reply src/db/resolvers/post_resolver.js]: Yeah I figured but I mean would it be possible to merge votePost into updatePost? or does it have to be kept separate

**Pull request merged**